File: CS/InfoPanels.Module.Web/InfoPanelViewControllerWeb.cs
using System.Web.UI;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.SystemModule;
using DevExpress.ExpressApp.Templates;
using DevExpress.ExpressApp.Web.Templates;

namespace InfoPanels.Module.Web {
    public interface IInfoPanelTemplateWeb : IFrameTemplate {
        Control PlaceHolder { get; }
    }
    public class InfoPanelViewControllerWeb : CustomizeTemplateViewControllerBase<IInfoPanelTemplateWeb> {
        LiteralControl literal;        
        protected override void AddControlsToTemplateCore(IInfoPanelTemplateWeb template) {
            if(literal == null) {
                literal = new LiteralControl();                                
            }
            template.PlaceHolder.Controls.Add(literal);
        }
        protected override void RemoveControlsFromTemplateCore(IInfoPanelTemplateWeb template) {
            template.PlaceHolder.Controls.Remove(literal);
            literal = null;
        }
        protected override void UpdateControls(View view) {
            UpdateControls();
        }
        protected override void UpdateControls(object currentObject) {
            UpdateControls();
        }
        void UpdateControls() {
            literal.Text = "The current View is " + View.Caption;
            if(View.CurrentObject != null) {
                literal.Text += "<br/>The current object is " + View.CurrentObject.ToString();
            }
        }
    }
}

File: VB/InfoPanels.Module.Web/InfoPanelViewControllerWeb.vb
Imports Microsoft.VisualBasic
Imports System.Web.UI
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.SystemModule
Imports DevExpress.ExpressApp.Templates
Imports DevExpress.ExpressApp.Web.Templates

Namespace InfoPanels.Module.Web
    Public Interface IInfoPanelTemplateWeb
    Inherits IFrameTemplate
        ReadOnly Property PlaceHolder() As Control
    End Interface
    Public Class InfoPanelViewControllerWeb
        Inherits CustomizeTemplateViewControllerBase(Of IInfoPanelTemplateWeb)
        Private literal As LiteralControl
        Protected Overrides Sub AddControlsToTemplateCore(ByVal template As IInfoPanelTemplateWeb)
            If literal Is Nothing Then
                literal = New LiteralControl()
            End If
            template.PlaceHolder.Controls.Add(literal)
        End Sub
        Protected Overrides Sub RemoveControlsFromTemplateCore(ByVal template As IInfoPanelTemplateWeb)
            template.PlaceHolder.Controls.Remove(literal)
            literal = Nothing
        End Sub
        Protected Overrides Overloads Sub UpdateControls(ByVal view As View)
            UpdateControls()
        End Sub
        Protected Overrides Overloads Sub UpdateControls(ByVal currentObject As Object)
            UpdateControls()
        End Sub
        Private Overloads Sub UpdateControls()
            literal.Text = "The current View is " & View.Caption
            If View.CurrentObject IsNot Nothing Then
                literal.Text &= "<br/>The current object is " & View.CurrentObject.ToString()
            End If
        End Sub
    End Class
End Namespace
